guide/english/algorithms/sieve-of-eratosthenes/index.md

```diff
@@ -0,0 +1,81 @@
+---
+title: sieve of Eratosthenes
+---
+
+## sieve of Eratosthenes
+Simple sieve algorithm for finding all prime numbers up to a designated limit.
+The earliest known reference to the sieve attributes it to Eratosthenes of
+Cyrene, a Greek mathematician.
+
+### How it works
+
+The algorithm operates on a set range of numbers starting at 1 through 'n'. It uses the following steps:
+
+  1. Create a list of all numbers in your range.
+  2. Remove 1 from the list, as it is not a prime number.
+  3. Circle the next number on the list.
+  4. Remove all multiples of the most recently circled number from the list.
+  5. Repeat steps 3 & 4 until you no longer have any numbers left in the list.
+
+  The first couple iterations look like this:
+  1.  * Circle 2
+      * Remove all multiples of 2 --- 4, 6, 8, ... 100
+  2.  * Circle 3
+      * Remove all multiples of 3 remaining on the list --- 9, 15, 21, ... 99
+
+### Example implementation in C
+
+C code: In this example our range is from 1 to n, where n == 100.
+
+```c
+
+#include <stdio.h>
+#include <stdlib.h>
+
+int main() {
+
+    int i, j;           // array indices
+    int *prime_array;   // array to mark if index is a prime number
```

Comment:
You have declared this as a pointer array `int *prime_array`, but I fail to understand the use of this in the code below as you are using this declared array as a normal array especially when you are hard-coding the input and therefore there is no need for using `malloc` to allocate the size, please either revert this to a normal array `int prime_array` or add an explanation as to why you are suggesting the use of a pointer array for the same and possibly add an option for a user_defined input.

P.S :- linked where in this all should happen(according to me) in the lines below.